File: app/src/main/java/fr/insapp/insapp/utility/TagTokenizer.java
package fr.insapp.insapp.utility;

import androidx.appcompat.widget.AppCompatMultiAutoCompleteTextView;

/**
 * Created by thomas on 28/02/2017.
 */

public class TagTokenizer implements AppCompatMultiAutoCompleteTextView.Tokenizer {

    @Override
    public int findTokenStart(CharSequence text, int cursor) {
        for (int i = cursor; i > 0; i--) {
            if (text.charAt(i - 1) == '@') {
                return i;
            }
        }

        return cursor;
    }

    @Override
    public int findTokenEnd(CharSequence text, int cursor) {
        return cursor;
    }

    @Override
    public CharSequence terminateToken(CharSequence text) {
        int i = text.length();

        while (i > 0 && text.charAt(i - 1) == ' ') {
            i--;
        }

        if (i > 0 && text.charAt(i - 1) == ' ') {
            return text;
        }

        return text + " ";
    }
}

File: app/src/main/java/fr/insapp/insapp/utility/TagTokenizer.kt
package fr.insapp.insapp.utility

import android.widget.MultiAutoCompleteTextView

/**
 * Created by thomas on 28/02/2017.
 * Kotlin rewrite on 03/09/2019.
 */

class TagTokenizer : MultiAutoCompleteTextView.Tokenizer {

    override fun findTokenStart(text: CharSequence, cursor: Int): Int {
        for (i in cursor downTo 1) {
            if (text[i - 1] == '@') {
                return i
            }
        }

        return cursor
    }

    override fun findTokenEnd(text: CharSequence, cursor: Int): Int {
        return cursor
    }

    override fun terminateToken(text: CharSequence): CharSequence {
        var i = text.length

        while (i > 0 && text[i - 1] == ' ') {
            i--
        }

        return if (i > 0 && text[i - 1] == ' ') {
            text
        } else "$text "
    }
}